docs/features/interceptors.md

```diff
@@ -0,0 +1,121 @@
+# Interceptors
+
+## Summary
+[summary]: #summary
+
+*Interceptors* are an experimental compiler feature. An *interceptor* is a method which can declaratively substitute a call to itself instead of a call to an *interceptable* method at compile time. This substitution occurs by having the interceptor declare the source locations of the calls that it intercepts. This provides a limited facility to change the semantics of existing code by adding new code to a compilation (e.g. in a source generator).
+
+```cs
+using System;
+using System.Runtime.CompilerServices;
+
+var c = new C();
+c.InterceptableMethod(1); // (L1,C1): prints `interceptor 1`
+c.InterceptableMethod(1); // (L2,C2): prints `other interceptor 1`
+c.InterceptableMethod(1); // prints `interceptable 1`
+
+class C
+{
+    [Interceptable]
+    public void InterceptableMethod(int param)
+    {
+        Console.WriteLine($"interceptable {param}");
+    }
+}
+
+// generated code
+static class D
+{
+    [InterceptsLocation("Program.cs", line: /*L1*/, character: /*C1*/)] // refers to the call at (L1, C1)
+    public static void InterceptorMethod(this C c, int param)
```

Comment:
>public static void InterceptorMethod(this C c, int param) [](http://example.com/codeflow?start=4&length=57)

I think the interceptor should match the ref-ness of `this` parameter of the method it is intercepting.